pkg/api/login_oauth.go

```diff
@@ -83,9 +83,9 @@ func (hs *HTTPServer) OAuthLogin(ctx *models.ReqContext) {
 		hashedState := hashStatecode(state, setting.OAuthService.OAuthInfos[name].ClientSecret)
 		cookies.WriteCookie(ctx.Resp, OauthStateCookieName, hashedState, hs.Cfg.OAuthCookieMaxAge, hs.CookieOptionsFromCfg)
 		if setting.OAuthService.OAuthInfos[name].HostedDomain == "" {
-			ctx.Redirect(connect.AuthCodeURL(state, oauth2.AccessTypeOnline))
+			ctx.Redirect(connect.AuthCodeURL(state, oauth2.AccessTypeOffline))
```

Comment:
Why is this change in here?

Reply:
@jprukner without this, the refresh token was not being sent to Grafana. Using this, Grafana stores the refresh token and automatically renews the access token in case it's expired, when calling the endpoint.